### ISSUE TITLE
SC Upgrade doesn't update transport type to correct one when non-MSMQ transport is used

### ISSUE THREAD

**Issue by SeanFeldman** [Particular/ServiceControl#574]
## Who's affected

Any user of ServiceControl upgrading to version 1.7 using SC Management Utility and using the following transports:

- Azure Service Bus
- Azure Storage Queues
- SqlServer
- RabbitMQ
- MSMQ (if defined explicitly)

## Symptoms

Upgrade process using SC Management Utility fails to restart updated SC, leaving SC unusable.
Exception during upgrade is:
```
Configuration of transport Failed. Could not resolve type '[transport type full name] from Setting
'TransportType'. Ensure the assembly is present and that type is correctly defined in settings
```

## Workaround

1. Stop SC (using the management tool or via services)
1. Locate your SC instance on disk (using management tool, select "Installation Path")
1. Open ServiceControl.exe.config file in a text editor
1. Locate appSettings section
1. Modify an entry with a `ServiceControl/TransportType` from by adding `Transport` to the type name (first part of value). 
  Example for ASB: `NServiceBus.AzureServiceBus, NServiceBus.Azure.Transports.WindowsAzureServiceBus` would become `NServiceBus.AzureServiceBusTransport, NServiceBus.Azure.Transports.WindowsAzureServiceBus`
1. Start SC

## Steps to repoduce:

1. Have SC earlier than version 1.7 configured to use ASB
![01 - sc 1 6 3 with asb running](https://cloud.githubusercontent.com/assets/1309622/10801811/e8b2e726-7d7e-11e5-92d0-f2f4b704ba55.PNG)

1. Perform upgrade
![02 - upgrading](https://cloud.githubusercontent.com/assets/1309622/10801820/f5ff8808-7d7e-11e5-9419-544b345cf34f.PNG)

1. Upgrade finished, but SC cannot start
![03 - upgrade fails with exception](https://cloud.githubusercontent.com/assets/1309622/10801921/628a357c-7d7f-11e5-9033-6319aa1b6f17.PNG)
(SC status)
![04 - stuck with 1 7 on asb not working](https://cloud.githubusercontent.com/assets/1309622/10801931/6b4a7c1c-7d7f-11e5-8ed0-f20f4ea02299.PNG)

The issue is with TransportType appSetting
![05 - cant modify transport type](https://cloud.githubusercontent.com/assets/1309622/10801954/89d769b0-7d7f-11e5-96dc-c0eca45900f7.PNG)

**Comment by johnsimons** [comment]
@SeanFeldman is this for all transports or just azure ones ?

**Comment by SeanFeldman** [comment]
@johnsimons I've tried on ASB. This will happen for all transports (MSMQ is in question) since from NSB v4 to v5 all transports changed their name.

**Comment by gbiellem** [comment]
Adding a new service instance worked correctly but upgrades did not update the config with the new transport type.  The types change between 1.6.3 and 1.7 as SC was changed to NSB v5.

The upgrade code in V1.7.1 correclty sets the type on upgrade and will fix a failed 1.7 upgrade as well.
